test/controllers/HomeControllerSpec.scala

```diff
@@ -0,0 +1,118 @@
+      val result = controller.presentWithdrawApplication()(fakeRequest)
+      status(result) must be(OK)
+      val content = contentAsString(result)
+      content must include("<title>Withdraw your application")
+      content must include(s"""<span class="your-name" id="bannerUserName">${currentCandidate.user.preferredName.get}</span>""")
+    }
+  }
+
+  "withdrawApplication" should {
+    "display withdraw form when the form was submitted invalid" in new TestFixture {
+      val Request = fakeRequest.withFormUrlEncodedBody(WithdrawApplicationFormExamples.OtherReasonInvalidNoReasonFormUrlEncodedBody: _*)
+
+      val result = controller.withdrawApplication()(Request)
+
+      status(result) must be(OK)
+      val content = contentAsString(result)
+      content must include(routes.HomeController.withdrawApplication().url)
+      content must include ("Select a reason for withdrawing your application")
+    }
+
+    "display dashboard with error message when form is valid but cannot withdraw" in new TestFixture {
+      val Request = fakeRequest.withFormUrlEncodedBody(WithdrawApplicationFormExamples.ValidFormUrlEncodedBody: _*)
+      when(mockApplicationClient.withdrawApplication(eqTo(currentApplicationId),
+        eqTo(WithdrawApplicationExamples.Valid))(any[HeaderCarrier])).thenReturn(Future.failed(new CannotWithdraw))
+
+      val result = controller.withdrawApplication()(Request)
+
+      status(result) must be(SEE_OTHER)
+      redirectLocation(result) must be(Some(routes.HomeController.present().url))
+      flash(result).data must be (Map("danger" -> "We can't find an application to withdraw"))
+    }
+
+    "display dashboard with withdrawn success message when withdraw is successful" in new TestFixture {
+      val Request = fakeRequest.withFormUrlEncodedBody(WithdrawApplicationFormExamples.ValidFormUrlEncodedBody: _*)
+      when(mockApplicationClient.withdrawApplication(eqTo(currentApplicationId),
+        eqTo(WithdrawApplicationExamples.Valid))(any[HeaderCarrier])).thenReturn(Future.successful(()))
+      when(mockApplicationClient.getApplicationProgress(eqTo(currentApplicationId))(any[HeaderCarrier]))
+        .thenReturn(Future.successful(ProgressResponseExamples.Submitted))
+
+      val Candidate = CachedData(currentCandidateWithApp.user, Some(currentCandidateWithApp.application))
+      val UpdatedApplication = currentCandidateWithApp.application
+        .copy(applicationStatus= ApplicationStatus.WITHDRAWN,  progress = ProgressResponseExamples.WithdrawnAfterSubmitted)
+      val UpdatedCandidate = currentCandidate.copy(application = Some(UpdatedApplication))
+      when(mockUserService.save(eqTo(UpdatedCandidate))(any[HeaderCarrier])).thenReturn(Future.successful(UpdatedCandidate))
+
+      val result = controller.withdrawApplication()(Request)
+
+      status(result) must be(SEE_OTHER)
+      redirectLocation(result) must be(Some(routes.HomeController.present().url))
+      //scalastyle:off
```

Comment:
This needs to be more specific e.g. // scalastyle:off line.length

or you can split the line across multiple lines like:

"this all appears" +
"on the same line" +
"in the app"